modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMIndexerImpl.java

```diff
@@ -429,13 +429,28 @@ else if (value instanceof Object[]) {
 
 						sb.append(stringArray);
 					}
-					else {
-						if (type.equals(DDMFormFieldTypeConstants.RICH_TEXT)) {
-							valueString = _htmlParser.extractText(valueString);
-						}
+					else if (type.equals(DDMFormFieldTypeConstants.RICH_TEXT)) {
+						valueString = _htmlParser.extractText(valueString);
 
 						sb.append(valueString);
 					}
+					else if (type.equals(
+								DDMFormFieldTypeConstants.DOCUMENT_LIBRARY) ||
+							 type.equals(DDMFormFieldTypeConstants.IMAGE) ||
+							 type.equals(
+								 DDMFormFieldTypeConstants.JOURNAL_ARTICLE) ||
+							 type.equals(
+								 DDMFormFieldTypeConstants.LINK_TO_LAYOUT)) {
+
+						JSONObject jsonObject =
+							JSONFactoryUtil.createJSONObject(valueString);
+
+						if ((jsonObject != null) && jsonObject.has("title")) {
```

Comment:
@rolandpakai what happens if the `jsonObject` doesn't have the `title` key? Also, did you check if all these types have a `title`?